src/components/UserInfo.vue

```diff
@@ -156,82 +164,84 @@ export default {
+        this.selectedClientId
+      ).catch(e => {
+        console.log(e);
+      });
     },
 
     getUserAvailableClientRoles: function() {
-      UsersRepository.getUserAvailableClientRoles(
+      return UsersRepository.getUserAvailableClientRoles(
         this.user.id,
         this.selectedClientId
-      )
-        .then(response => {
-          this.availableClientRoles.push(...response.data);
-        })
-        .catch(e => {
-          console.log(e);
-        });
+      ).catch(e => {
+        console.log(e);
+      });
     },
 
     getUserEffectiveClientRoles: function() {
-      UsersRepository.getUserEffectiveClientRoles(
+      return UsersRepository.getUserEffectiveClientRoles(
         this.user.id,
         this.selectedClientId
-      )
-        .then(response => {
-          this.effectiveClientRoles.push(...response.data);
-          this.selectedRoles.push(...response.data);
-        })
-        .catch(e => {
-          console.log(e);
-        });
+      ).catch(e => {
+        console.log(e);
+      });
     },
 
     updateUserClientRoles: function() {
-      //If in effective but not selected DELETE
-      var rolesToDelete = this.effectiveClientRoles.filter(
+      //If in roles but not selected DELETE
+      var rolesToDelete = this.clientRoles.filter(
         value => !this.selectedRoles.includes(value)
       );
-
-      //If in available and selected ADD
-      var rolesToAdd = this.availableClientRoles.filter(value =>
+      //If in roles and selected ADD
+      var rolesToAdd = this.clientRoles.filter(value =>
         this.selectedRoles.includes(value)
       );
 
       this.successMessage = "";
       this.errorMessage = "";
 
-      //TODO refactor this
-      UsersRepository.deleteUserClientRoles(
-        this.user.id,
-        this.selectedClientId,
-        rolesToDelete
-      )
-        .then(response => {
-          console.log(response);
-
-          return UsersRepository.addUserClientRoles(
-            this.user.id,
-            this.selectedClientId,
-            rolesToAdd
-          );
-        })
-        .then(response => {
-          console.log(response);
-
-          this.successMessage =
+      Promise.all([
+        UsersRepository.deleteUserClientRoles(
+          this.user.id,
+          this.selectedClientId,
+          rolesToDelete
+        ),
+        UsersRepository.addUserClientRoles(
+          this.user.id,
+          this.selectedClientId,
+          rolesToAdd
+        )
+      ]).then(() => {
+        this.getUserClientRoles();
+        this.successState = true;
+        this.errorState = false;
+        this.successMessage =
             this.successMessage + "Roles Updated Successfully ";
-
-          this.getUserClientRoles();
-        })
-        .catch(error => {
+        window.scrollTo(0, 0);    
+      }).catch(error => {
           this.errorMessage = this.errorMessage + "Error Updating Roles";
           console.log(error);
-        });
+      });
```

Comment:
UserInfo is only 250 lines long, but I wish could break up the functionality a little. As far as I can see the roles code and the user details code has zero `method` and `data` overlap: that is, they don't use any of the same methods or data. I thought they both used the `user` object, but looking again I don't think they even do that: it uses the `userid` in the path, but never actually GETs or POSTs to that resource.

Anyhow, I wouldn't reject a pull request for this, but if we have time I think we should look into it.

Reply:
Agreed, but will do this as a separate piece of work.